src/asynchro.rs

```diff
@@ -416,17 +419,26 @@ where
 
         // store last index for next iteration
         self.last_index = idx - self.chunk_size as f64;
-        for chan in used_channels.iter() {
-            //for w in wave_out.iter_mut() {
-            wave_out[*chan].truncate(n);
+        for (chan, active) in self.channel_mask.iter().enumerate() {
+            if *active {
+                wave_out[chan].truncate(n);
+            }
         }
+        #[cfg(debug_assertions)]
```

Comment:
I'm still unsure about this way of disabling logging. I think a Cargo feature would probably be better. The logging is useful if you're working on Rubato, but not so much for application developers using Rubato.

Reply:
My experience is that too many Cargo features can cause confusion, so I'd like to only use them when really needed. 
Anyone who wants to silence Rubato can do that by setting different logging levels for different modules: https://rust-lang-nursery.github.io/rust-cookbook/development_tools/debugging/config_log.html

Reply:
Does turning off logging for a module disable the allocation of the String?

Reply:
I am using env_logger but it would be annoying to have to always disable a library's logging.

Reply:
Don't know, but in debugging mode things are running so slow anyway that some extra allocations shouldn't make any difference.

Reply:
In a release build there are no logging calls in the realtime-critical parts.

Reply:
> Don't know, but in debugging mode things are running so slow anyway that some extra allocations shouldn't make any difference.

You should not assume that as a library author. I set the optimization level to 2 for my debug builds.

Reply:
[log_enabled!](https://docs.rs/log/0.4.14/log/macro.log_enabled.html) will skip allocating the log String if logging is disabled for that message.

Reply:
Looks like a normal log call like `debug!()`also skips it: https://docs.rs/log/0.4.14/src/log/macros.rs.html#31
The argument ends up inside the if-statement so should never be allocated if the level is disabled.

Reply:
Is there a way for a module to set its own default log level?

Reply:
I was looking at how others do this, and found a quite nice solution in `rand`. It defines new logging macros that basically decorates the normal ones with a conditional, checking the Cargo feature `log`:
https://github.com/rust-random/rand/blob/master/src/lib.rs#L62

This requires adding a Cargo feature, but then the code itself gets nice and clean. I think I'll reverse it compared to `rand` and add a feature that disables the logging instead.

Reply:
Done! But not exactly as I wrote before. It's actually discouraged to have features disable functionality so I did it as in `rand` and added a `log` feature. This is enabled by default.

Reply:
Oh, that's a nice hack! I agree that Cargo features should be additive. I'd suggest to disable the feature by default though. Either way, the documentation should explicitly state that the feature allocates and shouldn't be used in realtime code.

Reply:
There aren't that many loggers that support async logging (really important if you want to support running in the strange Windows terminal for example!). I have found only `slog` and `flexi_logger`. I started with `slog` but wasn't really happy with it. It's too big (for me) and complicated to use. But `flexi_logger` is really great.